packages/freezed/lib/src/templates/concrete_template.dart

```diff
@@ -94,6 +94,31 @@ ${copyWith.abstractCopyWithGetter}
 ''';
   }
 
+  String get _implementations {
+    final implementationTypes = constructor.decorators
+        .where((element) => element.startsWith('@Implements'))
+        .map((element) {
+          // the fromString constructor contains commas
+          final offset = element.contains('fromString') ? 2 : 1;
```

Comment:
Instead of parsing the source code, we can read the content of the decorator and read stringType.